crates/sui-graphql-rpc/src/extensions/logger.rs

```diff
@@ -140,29 +143,76 @@ impl Extension for LoggerExtension {
                             }
                         }
                     }
-
-                    error!(
-                        target: "async-graphql",
-                        "[Response] path={} message={}", path, err.message,
-                    );
+                    if let Some(async_graphql_value::ConstValue::String(val)) = error_code {
```

Comment:
if you add a code::UNKNOWN enum value this if block/test will be easier to reason through and become shorter.